src/Core/Domain/Address/Command/AbstractEditAddressCommand.php

```diff
@@ -332,23 +333,23 @@ public function setAddress2(string $address2): self
     }
 
     /**
-     * @return StateId|null
+     * @return StateIdInterface|null
      */
-    public function getStateId(): ?StateId
+    public function getStateId(): ?StateIdInterface
     {
         return $this->stateId;
     }
 
     /**
-     * @param int $stateId
+     * @param StateIdInterface $stateId
      *
      * @return self
      *
      * @throws StateConstraintException
      */
-    public function setStateId(int $stateId): self
+    public function setStateId(StateIdInterface $stateId): self
```

Comment:
```suggestion
    public function setStateId(int $stateId): self
```

Reply:
We still use primitives as the parameter for setter, but the VO is created in the setter Like in this example:
https://github.com/PrestaShop/PrestaShop/blob/1976f437da2fcd3f793a7be016f774a438ca2a18/src/Core/Domain/Product/SpecificPrice/Command/EditProductSpecificPriceCommand.php#L322-L327

Reply:
copy that !